tests/shared_tests.py

```diff
@@ -0,0 +1,28 @@
+def run_save_load_train(happy, output_path, args, data_path, run_type):
```

Comment:
what is this file / function ???

Reply:
Test functions like "test_wp_save_load_train()" and "test_tc_save_load_eval()" call it to reduce code duplication